File: 0888-fair-candy-swap/0888-fair-candy-swap.java
class Solution {
    public int[] fairCandySwap(int[] a, int[] b) {
        Set<Integer> set=new HashSet();
        int[] ans=new int[2];

        int suma=0,sumb=0;
        for(int i=0;i<a.length;i++)
        suma+=a[i];
        for(int j=0;j<b.length;j++){
            sumb+=b[j];
            set.add(b[j]);
        }

        int finalAns=(suma+sumb)/2;
        for(int i=0;i<a.length;i++){
            int value=finalAns-suma+a[i];
            if(set.contains(value)){
                ans[0]=a[i];
                ans[1]=value;
                return ans;
            }
        }
    return ans;    
    }
}

File: 0888-fair-candy-swap/0888-fair-candy-swap.py
class Solution:
    def fairCandySwap(self, aliceSizes: List[int], bobSizes: List[int]) -> List[int]:
        setb=set()
        ans=[]
        suma=sum(aliceSizes)
        sumb=0
        for i in range(len(bobSizes)):
            sumb+=bobSizes[i]
            setb.add(bobSizes[i])
        finalsum=(suma+sumb)/2
        for c in range(len(aliceSizes)):
            value=finalsum-suma+aliceSizes[c]
            if value in setb:
                ans.append(aliceSizes[c])
                ans.append(value)
                return ans
        return ans